smdebug/core/modes.py

```diff
@@ -14,3 +14,16 @@ class ModeKeys(Enum):
 ALLOWED_MODE_NAMES = [x.name for x in ALLOWED_MODES]
 MODE_STEP_PLUGIN_NAME = "mode_step"
 MODE_PLUGIN_NAME = "mode"
+
+
+def str_to_mode_keys(s):
```

Comment:
this is being used only in should_save_global_step_for_mode, which doesn't have a caller. is this required?

Reply:
`should_save_global_step_for_mode` is called in aws-tf. See https://tiny.amazon.com/ljsp41dd/IsenLink for details.